overlord/snapstate/handlers.go

```diff
@@ -416,6 +425,14 @@ func (m *SnapManager) doDownloadSnap(t *state.Task, tomb *tomb.Tomb) error {
 
 	meter := NewTaskProgressAdapterUnlocked(t)
 	targetFn := snapsup.MountFile()
+
+	ctx := tomb.Context(nil)
+	st.Lock()
```

Comment:
we seem to be frobbing lock/unlock on  state a lot in this function. Mind moving this one up?